pkg/controller/controller.go

```diff
@@ -61,43 +81,91 @@ func NewController(client kubernetes.Interface, pvcInformer cache.SharedIndexInf
 }
 
 func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
-	defer c.queue.ShutDown()
+	defer c.pvcQueue.ShutDown()
 	glog.Infoln("Starting CDI controller loop")
 	if threadiness < 1 {
 		return fmt.Errorf("controller.Run: expected >0 threads, got %d", threadiness)
 	}
 	go c.pvcInformer.Run(stopCh)
+	go c.podInformer.Run(stopCh)
+
 	if !cache.WaitForCacheSync(stopCh, c.pvcInformer.HasSynced) {
-		return fmt.Errorf("controller.Run: Timeout waiting for cache sync")
+		return fmt.Errorf("controller.Run: Timeout waiting for pvc cache sync")
+	}
+	if !cache.WaitForCacheSync(stopCh, c.podInformer.HasSynced) {
+		return fmt.Errorf("controller.Run: Timeout waiting for pod cache sync")
 	}
 	glog.Infoln("Controller cache has synced")
 
 	for i := 0; i < threadiness; i++ {
-		go wait.Until(c.runWorkers, time.Second, stopCh)
+		go wait.Until(c.runPVCWorkers, time.Second, stopCh)
+		go wait.Until(c.runPodWorkers, time.Second, stopCh)
 	}
 	<-stopCh
 	return nil
 }
 
-func (c *Controller) runWorkers() {
-	for c.ProcessNextItem() {
+func (c *Controller) runPodWorkers() {
+	glog.Infoln("[=== DEBUG ===] Starting pod workers")
+	for c.ProcessNextPodItem() {
+	}
+}
+
+func (c *Controller) runPVCWorkers() {
+	for c.ProcessNextPvcItem() {
+	}
+}
+
+func (c *Controller) ProcessNextPodItem() bool {
+	glog.Infoln("[=== DEBUG ===] ProcessNextPodItem")
+	key, shutdown := c.podQueue.Get()
+	if shutdown {
+		return false
+	}
+	glog.Infoln("[=== DEBUG ===] dequeuing pod key")
+	defer c.podQueue.Done(key)
+
+	keyString, ok := key.(string)
+	if !ok {
+		return false
+	}
+
+	glog.Infof("[=== DEBUG ===] getting pod from store by key %q\n", keyString)
+	obj, ok, err := c.podInformer.GetIndexer().GetByKey(keyString)
+	if err != nil || !ok {
+		return false
 	}
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		return false
+	}
+	if err := c.processPodItem(pod); err == nil {
```

Comment:
I thought we want to Forget for success case and not forget if there was an error?

Reply:
right, err is nil here, so the process succeeded.  Forget the key so we don't keep requeueing it